package.json

```diff
@@ -3,12 +3,14 @@
   "version": "4.9.0",
   "description": "JavaScript library for StarkNet",
   "main": "dist/index.js",
-  "module": "dist/index.esm.js",
+  "module": "dist/index.mjs",
   "browser": "dist/index.global.js",
   "types": "dist/index.d.ts",
   "scripts": {
     "prepare": "npm run build && husky install",
     "build": "tsup",
+    "build:esm": "tsup --clean false --format esm --platform node",
```

Comment:
does this need to be node? wouldnt it work without the plattform parameter and work in browser and node?

Reply:
No, for the `esm` build the `platform` parameter is not necessary, any value will result in the same build. To go into a bit more detail, the `platform` parameter is necessary when bundling the code of the full dependency tree, specifically the `@noble/hashes` sub-dependency through `ethereum-cryptography`. The full bundling is done for the `iife` build, the `esm` and `cjs` builds keep only `import`/`require` references to the direct dependencies, so they don't go deeper than `ethereum-cryptography`.

I left it to have symmetry with `build:iife` and to make its status as an implicit default explicit (not providing any platform value is the same as providing `node`, [link](https://github.com/egoist/tsup/blob/48b3381b15562490870b0cf32f15252038958384/src/esbuild/index.ts#L130)).